src/main/java/org/opentripplanner/middleware/connecteddataplatform/AnonymizedTripRequest.java

```diff
@@ -50,24 +69,162 @@ public AnonymizedTripRequest() {
+                processLegCoordinates(itinerary.legs);
+                for (int i = 0; i < itinerary.legs.size(); i++) {
+                    Leg leg = itinerary.legs.get(i);
+                    AnonymizedLeg anonymizedLeg = new AnonymizedLeg();
+                    // Parameters for both transit and non transit legs.
+                    anonymizedLeg.distance = leg.distance;
+                    anonymizedLeg.duration = leg.duration;
+                    anonymizedLeg.startTime = leg.startTime;
+                    anonymizedLeg.endTime = leg.endTime;
+                    anonymizedLeg.mode = leg.mode;
+                    anonymizedLeg.transitLeg = leg.transitLeg;
+                    anonymizedLeg.fromStop = leg.from.stopId;
+                    anonymizedLeg.from = new Coordinates(leg.from.lat, leg.from.lon);
+                    anonymizedLeg.toStop = leg.to.stopId;
+                    anonymizedLeg.to = new Coordinates(leg.to.lat, leg.to.lon);
+                    if (Boolean.TRUE.equals(leg.transitLeg)) {
+                        // Parameters for a transit leg.
+                        anonymizedLeg.agencyId = leg.agencyId;
+                        anonymizedLeg.interlineWithPreviousLeg = leg.interlineWithPreviousLeg;
+                        anonymizedLeg.realTime = leg.realTime;
+                        anonymizedLeg.routeId = leg.routeId;
+                        anonymizedLeg.routeShortName = leg.routeShortName;
+                        anonymizedLeg.routeLongName = leg.routeLongName;
+                        anonymizedLeg.routeType = leg.routeType;
+                        anonymizedLeg.tripBlockId = leg.tripBlockId;
+                        anonymizedLeg.tripId = leg.tripId;
+                    } else {
+                        // Parameters for non transit leg.
+                        anonymizedLeg.rentedVehicle = leg.rentedVehicle;
+                    }
+                    itin.legs.add(anonymizedLeg);
+                }
+            }
+            anonymizedItineraries.add(itin);
+        }
+        return anonymizedItineraries;
+    }
+
+    /**
+     * Define the coordinates for all legs. If the leg is non-transit and is before the first transit leg or after
+     * the last transit leg the coordinates must be removed for privacy. Any transit legs or non-transit legs
+     * between the first and last transit legs, the coordinates can be provided.
+     */
+    private void processLegCoordinates(List<Leg> legs) {
+        int firstTransitLeg = getFirstTransitLeg(legs);
+        // No need to find the last transit leg if the first transit leg indicates that all legs are non-transit.
+        int lastTransitLeg = (firstTransitLeg == Integer.MAX_VALUE) ? Integer.MIN_VALUE : getLastTransitLeg(legs);
+        for (int i = 0; i <= legs.size() -1; i++) {
+            Leg leg = legs.get(i);
+            if (Boolean.TRUE.equals(!leg.transitLeg) && (i < firstTransitLeg || i > lastTransitLeg)) {
```

Comment:
I could but SonarLint suggests doing it this way because the Boolean value is a class and not a primitive. Happy to change this and others if you think it is cleaner?

Reply:
@binh-dam-ibigroup It's that a thumbs up to change or leave?!

Reply:
It's fine to follow SonarLint.